File: test/benchmark/cases/call_dense/case.java
class Main {
    static int stepA(int value) {
        return (value * 3 + 1) % 1009;
    }

    static int stepB(int value) {
        return (stepA(value) + stepA(value + 7)) % 1009;
    }

    static int stepC(int value) {
        return (stepB(value) * 5 + stepA(value - 3)) % 1009;
    }

    public static void main(String[] args) {
        var raw = System.getenv("CP_BENCH_INPUT");
        if(raw == null) {
            System.exit(1);
        }

        var count = Integer.parseInt(raw);
        var index = 0;
        var total = 0;
        while(index < count) {
            total = (total + stepC(index)) % 1000003;
            index += 1;
        }
        System.exit(total % 251);
    }
}

File: test/benchmark/cases/call_dense/case.cpp
#include <cstdlib>

auto step_a(int value) -> int
{
    return (value * 3 + 1) % 1009;
}

auto step_b(int value) -> int
{
    return (step_a(value) + step_a(value + 7)) % 1009;
}

auto step_c(int value) -> int
{
    return (step_b(value) * 5 + step_a(value - 3)) % 1009;
}

auto main() -> int
{
    auto const* raw = std::getenv("CP_BENCH_INPUT");
    if(raw == nullptr) {
        return 1;
    }

    auto const count = std::atoi(raw);
    auto index = 0;
    auto total = 0;
    while(index < count) {
        total = (total + step_c(index)) % 1000003;
        index += 1;
    }
    return total % 251;
}
